lib/ansible/modules/system/btrfs_quota.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Pawel Szatkowski <pszatkowski.byd@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: btrfs_quota
+
+author:
+    - Pawel Szatkowski (@pszatkowski) <pszatkowski.byd@gmail.com>
+
+version_added: '2.8'
+
+short_description: Manage btrfs quota
+
+description:
+    - This module manage btrfs quota
+
+options:
+    state:
+        description:
+            - Enable or disable quota for btrfs filesystem
+        choices: [enabled, disabled]
```

Comment:
would `present` and `absent` be good state names? i.e., is this something that is turned on or off per-filesystem, or is it a set of policies that can be configured on a more granular level? (for example some policies present, others absent) (I'm not familiar w/ these quotas)

Reply:
otherwise, many modules have a separate 'enabled' true/false, like enabling a service but not (necessarily) starting it

Reply:
@jamescassell Quota support is turned on/off per-filesystem. It must be enabled first, otherwise it's not possible to set limits. 

Regarding to `present ` and `absent ` states, well that's a good point. It's something that current code is not doing. It doesn't create or delete qgroups, because 0-level qgroups are automatically created/deleted for each btrfs subvolume and I wrote this module for that use case - I wanted have controll over subvolumes data.

However qgroup mechanism is much more richfull. It allows to create quota groups hierrarchy in parent-child relation. It means that you can define higher level qgroups (1-level, 2-level, etc.) that contains lower level qgroups (childs). Limit set on parent also limits childs. It's well explained in this [withepaper](http://sensille.com/qgroups.pdf).

Nevertheless the most frequent use case is to limit subvolumes data only, but if module should use whole potential of qgroups then indeed `present` and `absent` states should be added as well as `parent` option for setting relation to parent qgroup.